modules/ROOT/pages/dataweave-cookbook-decode-encode.adoc

```diff
@@ -0,0 +1,158 @@
+----
+<http:listener-config name="HTTP_Listener_config"
+                      doc:name="HTTP Listener config" >
+    <http:listener-connection
+                            host="0.0.0.0"
+                            port="8081" />
+</http:listener-config>
+<file:config name="File_Config"
+             doc:name="File Config" >
+    <file:connection workingDir="/Users/me/Downloads" />
+</file:config>
+<flow name="encode-decode-base64">
+    <http:listener doc:name="Listener"
+                 path="/transform"
+                 config-ref="HTTP_Listener_config"/>
+    <file:read doc:name="Read"
+             config-ref="File_Config"
+             path="pdf-test.pdf" />
+    <ee:transform doc:name="Transform Message">
+        <ee:message>
+          <ee:set-payload><![CDATA[%dw 2.0
+import * from dw::core::Binaries
+output text/plain
+---
+toBase64(payload as Binary)]]></ee:set-payload>
+        </ee:message>
+    </ee:transform>
+    <ee:transform doc:name="Transform Message" >
+        <ee:message >
+          <ee:set-payload ><![CDATA[%dw 2.0
+import * from dw::core::Binaries
+output application/octet-stream
+---
+fromBase64(payload as String) as Binary]]></ee:set-payload>
+        </ee:message>
+    </ee:transform>
+    <file:write doc:name="Write"
+              path="/Users/me/pdf-test-result.pdf"/>
+    <set-payload
+              value='#["PDF file created in the directory specified in the Write operation."]'
+              doc:name="Set Payload" />
+</flow>
+----
++
+. Place a PDF file of your choice into a local directory on your machine.
+. Correct the input and output directories of the Read and Write operations:
++
+* In the Read operation, change the *File Path* value `/Users/me/Downloads" to the location of your PDF file.
+* In the Write operation, change the *Path* value `/Users/me/pdf-test.pdf"` to the location and file name that you want to output the file.
+. Run the project in Studio.
+. Load the `0.0.0.0:8081/transform` into a browser.
++
+This action creates a PDF file in the specified location and generates a separate `transform` file with the message in the Set Payload component, `PDF file created in the directory specified in the Write operation.`
+
+[[ex01]]
+== Encode a PDF File to Base64
+
+You can simulate the transformation in the Read operation from the <<xml_config, XML configuration>>. This example performs Base64 encoding on a PDF file. More specifically, it converts a Base64 file stream (`application/octet-stream`) to Binary and outputs the Binary in the `text/plain` format.
+
+The example uses the following functions:
+
+*  `readUrl` from the `Core (dw::Core)` module to input a PDF file.
++
+The example assumes that the PDF file is in the `src/main/resources` directory of a Mule project in Anypoint Studio. It provides the name of the PDF file and specifies the MIME type as `application/octet-stream`.
+* `fromBase64` from the `dw::core::Binaries` module to transform a Base64 string to a binary value.
```

Comment:
Shouldn't this go beneath the other item?

Reply:
done